restapi/src/main/java/io/stargate/web/docsapi/service/query/search/weigth/impl/UserOrderWeightResolver.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright The Stargate Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package io.stargate.web.docsapi.service.query.search.weigth.impl;
+
+import com.bpodgursky.jbool_expressions.Expression;
+import io.stargate.web.docsapi.service.query.FilterExpression;
+import io.stargate.web.docsapi.service.query.search.weigth.ExpressionWeightResolver;
+import java.util.Collection;
+import java.util.List;
+
+/** The {@link ExpressionWeightResolver} that respects the user oder of the expressions. */
+public class UserOrderWeightResolver implements ExpressionWeightResolver<FilterExpression> {
+
+  private final List<Expression<FilterExpression>> order;
+
+  public UserOrderWeightResolver(List<Expression<FilterExpression>> order) {
+    this.order = order;
+  }
+
+  /** {@inheritDoc} */
+  @Override
+  public int compare(FilterExpression o1, FilterExpression o2) {
+    int defaultCompare = ExpressionWeightResolver.super.compare(o1, o2);
+    if (defaultCompare != 0) {
+      return defaultCompare;
+    } else {
+      return Integer.compare(indexOrMax(order, o1), indexOrMax(order, o2));
```

Comment:
nit: We could just assign index values to expressions at parse time, I guess.

Reply:
I might be worth introducing a "selectivity" property for filters (as @EricBorczuk suggested privately) at least for internal use cases. We do not have to expose it to the user as an explicit parameter, but could derive it from the filter index for now.

I guess it might make dealing with filter order easier for us internally. We should not have to worry about the order of child elements inside boolean expressions, but we could use this "selectivity" value when we have to choose the filter for the "next" query at each level during the resolution process. WDYT?

For parallel execution we can run queries independently regardless of their relative selectivity.

Reply:
I like this idea very much. Then the weight can become static again.

Reply:
Done.